coa-helm/templates/cromwell.yaml

```diff
@@ -0,0 +1,168 @@
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: {{ include "app.fullname" . }}-cromwell-deploy
+  labels:
+    {{- include "app.labels" . | nindent 4 }}
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      {{- include "app.cromwell.selectorLabels" . | nindent 6 }}
+  strategy:
+    type: Recreate
+  template:
+    metadata:
+      labels:
+        {{- include "app.cromwell.selectorLabels" . | nindent 8 }}
+    spec:
+      containers:
+        - args:
+            -  java -Djava.io.tmpdir=/cromwell-tmp/ -Dconfig.file=/configuration/cromwell/cromwell-application.conf -jar /app/cromwell.jar server
+          command:
+            - /bin/sh
+            - -c
+          image: {{ .Values.images.cromwell }}
+          name: cromwell
+          stdin: true
+          tty: true
+          ports:
+            - containerPort: {{ .Values.service.cromwellPort }}
+          resources: {}
+          volumeMounts:
+            - mountPath: /cromwell-tmp
+              name: cromwell-tmp-claim
+            - mountPath: /configuration
+              name: configuration-claim1
+            - mountPath: /cromwell-executions
+              name: cromwell-executions-claim1
+            - mountPath: /cromwell-workflow-logs
+              name: cromwell-workflow-logs-claim1
+            - mountPath: /workflows
+              name: workflows-claim1
+            - mountPath: /inputs
+              name: inputs-claim1
+            - mountPath: /outputs
+              name: outputs-claim1
+            - mountPath: /configuration/cromwell
+              name: cromwell-config-volume
+      restartPolicy: Always
+      volumes:
+        - name: cromwell-tmp-claim
+          persistentVolumeClaim:
+            claimName: cromwell-tmp-claim
+        - name: configuration-claim1
+          persistentVolumeClaim:
+            claimName: configuration-claim1
+        - name: cromwell-executions-claim1
+          persistentVolumeClaim:
+            claimName: cromwell-executions-claim1
+        - name: cromwell-workflow-logs-claim1
+          persistentVolumeClaim:
+            claimName: cromwell-workflow-logs-claim1
+        - name: workflows-claim1
+          persistentVolumeClaim:
+            claimName: workflows-claim1
+        - name: inputs-claim1
+          persistentVolumeClaim:
+            claimName: inputs-claim1
+        - name: outputs-claim1
+          persistentVolumeClaim:
+            claimName: outputs-claim1
+        - name: cromwell-config-volume
+          configMap:
+            name: {{ include "app.fullname" . }}-cromwell-config
```

Comment:
this block defines a volume based on the configMap with the name specified

Reply:
Huh, so when we mount a config map, Helm understands that to mean we create file in the mounted directory for each item in the map, and insert that item's value into it? Or is there something here that's telling it to do that as opposed to something else?

Reply:
yes! 

though I think it's a Kubernetes feature (you don't have to be using Helm, necessarily)

Reply:
more info here: https://kubernetes.io/docs/tasks/configure-pod-container/configure-pod-configmap/